src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -251,7 +251,7 @@ synchronized public void resourceOffers(SchedulerDriver driver, List<Offer> offe
           pendingOffers.put(offer.getId(), offer);
       }
         
-      if (!initializingCluster) {
+      if (!initializingCluster && pendingOffers.size() >= 3) {
```

Comment:
Don't hard code the number 3, please use the constants number of name nodes number of journal nodes, etc. They are used throughout the class